security-force-monitor/sfm-cms
# Clean up translations for all data entry views

## Overview

This PR represents a comprehensive audit of the data entry views to confirm that they're being translated properly.

Connects #580 

### Notes

For some basic background on how translations work in Django, see the `Overview` section of [Django's Translation docs](https://docs.djangoproject.com/en/2.2/topics/i18n/translation/).

There are a lot of templates in the project that I didn't consider as part of this PR; you may notice that these templates occasionally display untranslated strings. The problem here is that this project has developed a lot of cruft over the years, and many of the templates aren't used anywhere. My strategy for identifying templates and views that needed to be updated was to follow the CRUD links for every available entity type, adjusting the views and templates that were implicated by each page, and ignoring the rest. In a few cases, I stumbled on templates that I was able to confirm weren't being used anywhere, and I went ahead and deleted those.

## Testing Instructions

Testing this PR is a bit of a challenge, since we'll need to actually provide the relevant translations before we'll be able to see the new tagged strings in this PR show up in their translated form. Instead, I think the important task is to **confirm that all pages continue to load sensibly** (i.e. that no strings on the page show up with artifacts indicating improper translation, including Django template fragments like `%}`):

- Select the Spanish translation of the site
- For each entity type in Personnel, Units, Incidents, Sources, and Locations:
    - Navigate to the Create view and confirm it looks good
    - Navigate to a detail page for a record, Choose "Edit," and for each possible editing page confirm that it looks good
        - If the editing page supports deletion (e.g. Unit Compositions can be deleted), attempt to delete a record and confirm that the confirmation page looks good
    - If the entity type supports deletion (e.g. Locations can be deleted) attempt to delete a record and confirm that the confirmation page looks good

This work, including the review of the code in this PR, doesn't have to be particularly thorough. We'll do a more thorough test of this PR when the translators can provide translations all of the new strings tagged here, at which point we'll be able to tell visually if a particular string escaped our notice.
